src/client/fuse_ll.cc

```diff
@@ -163,6 +167,39 @@ class CephFuse::Handle {
   struct fuse_args args;
 };
 
+static int get_mount_status(const char *path, bool &is_mounted)
+{
+  FILE *mtab = NULL;
+  struct mntent *mnt;
+  int err = 0;
+
+  is_mounted = false;
+
+  // get mount information from /etc/mtab
+  mtab = setmntent(_PATH_MOUNTED, "r");
+  if (mtab == NULL) {
+    err = errno;
+    derr << "fuse_ll: get_mount_status: cannot open " << _PATH_MOUNTED
+      << ". Failed with error " << cpp_strerror(err) << dendl;
+    return err;
+  }
+
+  // check whether path is already mounted for ceph_fuse
+  while ((mnt = getmntent(mtab)) != NULL) {
```

Comment:
This obviously works, but it does rely on /etc/mtab being present. If it's not then you won't be able to mount.

Before groveling around in /etc/mtab, it may be better to test whether the dentry in question is already an existing mountpoint. The usual method for that is to statx the dentry you're mounting on, and its parent and see whether their stx_dev_* values are the same. If they aren't then you know it's a mountpoint.

Reply:
Actually, what may be better than scraping mtab like this is to query for some xattr that you know will be present on ceph-fuse but generally not on other filesystems. I know that ceph-fuse supports some vxattrs, so maybe we can use one of those.

Maybe do a getxattr of `ceph.client_id` ? If it comes back with an `ENODATA` error then you know it's not cephfs and can overlay it?

Reply:
Alternately if you just want to see whether the underlying fs on a mountpoint is FUSE, then you can use statfs(2) on the mountpoint and see if the f_type field is FUSE_SUPER_MAGIC:

`#define FUSE_SUPER_MAGIC 0x65735546`

Unfortunately, I don't think the MAGIC values are in a header anywhere, so you'll have to define the value yourself in the code.

Reply:
Actually most of them are in `/usr/include/linux/magic.h`, but FUSE is not in there (and neither is kernel cephfs!). That's probably a bug. I'll plan to submit a patch upstream for that soon. In the meantime, just do something like this in the code, and add the include for magic.h before the definition:

```
#ifndef FUSE_SUPER_MAGIC
#define FUSE_SUPER_MAGIC 0x65735546
#endif
```

If it shows up in the future, we'll end up preferring the value in magic.h.

Reply:
Using getxattr and checking ceph.client_id attribute to detect ceph fuse.
I will FUSE_SUPER_MAGIC check if I only need to check whether mountpoint is of type FUSE.